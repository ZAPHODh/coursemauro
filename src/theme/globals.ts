'use client'

import { Montserrat } from 'next/font/google'
import { createGlobalStyle } from 'styled-components'
import { css } from 'styled-components'

const montserrat = Montserrat({
    subsets: ['latin'],
    fallback: ['Georgia', 'ui-serif', 'serif'],
    weight: ['300', '700'],
})

export const GlobalStyles = createGlobalStyle`${() => css`
    html {
        font-family: ${montserrat.style.fontFamily};
        max-width: 100vw;
    }
    ,
    main {
        max-width: 100vw;
    }
    body {
        padding: 0;
        margin: 0;
        overflow: hidden;
    }
    ,
    input {
        font-family: inherit;
    }
    ,
    a {
        color: inherit;
    }
    ,
    * {
        box-sizing: border-box;
    }
    ,
    button {
        font-family: inherit;
    }
`}`
