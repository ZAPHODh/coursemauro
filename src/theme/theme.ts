import { DefaultTheme } from 'styled-components'

export const theme: DefaultTheme = {
    colors: {
        white: '#ffffff ',
        darker: '#0B0B0D',
        dark: '#181619',
        normalDark: '#181B22',
        brightDark: '#272A31',
        gold: '#A76B09',
        brightGold: '#DDA74F',
    },
}
