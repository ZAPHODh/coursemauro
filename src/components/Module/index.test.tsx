import { render } from '@testing-library/react'
import { Module } from '.'
import { ThemeClient } from '../../theme/ThemeClient'

describe('Module', () => {
    it('should render the Module', () => {
        const { getByRole } = render(
            <ThemeClient>
                <Module name="test" number={1} />
            </ThemeClient>
        )

        const component = getByRole('heading')

        expect(component).toBeInTheDocument()
    })
    it('should render with correct name params', async () => {
        const { getByRole } = render(
            <ThemeClient>
                <Module name="test" number={1} />
            </ThemeClient>
        )
        const heading = getByRole('heading')
        expect(heading).toHaveTextContent('test')
    })
    it('should render with correct number params', async () => {
        const { getByText } = render(
            <ThemeClient>
                <Module name="test" number={1} />
            </ThemeClient>
        )
        const mod = getByText('Módulo 1')
        expect(mod).toBeInTheDocument()
    })
    it('should render with correct number params', async () => {
        const { getByText } = render(
            <ThemeClient>
                <Module name="test" number={1} />
            </ThemeClient>
        )
        const mod = getByText('Módulo 1')
        expect(mod).toBeInTheDocument()
    })
    it('should take a snapshot of the document', async () => {
        const { getByRole } = render(
            <ThemeClient>
                <Module name="test" number={1} />
            </ThemeClient>
        )

        const component = getByRole('heading')

        expect(component).toMatchSnapshot()
    })
})
