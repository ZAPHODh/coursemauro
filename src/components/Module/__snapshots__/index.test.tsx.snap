// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module should take a snapshot of the document 1`] = `
<h2
  class="sc-9eb0259e-0 kSMpJi"
>
  test
</h2>
`;
